testpmd/hooks/pre-run.yml

```
@@ -41,12 +37,56 @@
     run_migration_test: false
     mac_workaround_enable: false
 
-- name: "Create CNF Namespace"
+# Expecting variables in an array on this form
+# sriov_network:
+#  - name: intel-numa0-net1
+#    vlan: 460
+#    policy: intel-numa0-policy1
+#    resource: intel_numa0_res1
+#    numvfs: 16
+#    mtu: 9000
+#    device_type: vfio-pci
+#    devices:
+#      - ens2f1#0-15
+#    target_namespace: example-cnf
+# or
+#  - policy: intel-dpdk-node-policy
+#    resource: intelnics
+#    numvfs: 4
+#    vendor: 8086
+#    deviceid: 1592
+#    devices:
+#      - ens4f0
+#    rootdevices:
+#      - "0000:d8:00.0"
+#    nodeselector: '"node.openshift.io/nic_type": "intel810"'
+- name: "Create SriovNetworkTargetNamespace {{ sriov['target_namespace'] }}"
   k8s:
     api_version: v1
     kind: Namespace
     state: present
-    name: "{{ cnf_namespace }}"
+    name: "{{ sriov['target_namespace'] }}"
+  loop: "{{ sriov_network }}"
+  loop_control:
+    loop_var: sriov
+  when:
+    - sriov_network is defined
```

I like the reuse of vars, but I think is going to be better to keep it simple for the namespace:
```suggestion
    name: "{{ cnf_namespace }}"
```

---

Yes, let's keeping it simple, I updated this. I also updated the sriov-network template to use the fact, and remove the target_template variable from the sriov_network array